src/plugins/intel_gpu/src/graph/graph_optimizer/remove_redundant_reorders.cpp

```diff
@@ -265,6 +265,35 @@ void remove_redundant_reorders::run(program& p) {
         auto o_layout = r_node.get_output_layout();
         auto i_layout = r_node.get_dependency(0).get_output_layout();
 
+        // Check redundant 1d reorder
+        if (i_layout.data_type == o_layout.data_type && i_layout.count() == o_layout.count()) {
+            // Check feature size of byxf input fits to output's inner block size
+            if (i_layout.format == format::byxf &&
+                ((o_layout.format == format::b_fs_yx_fsv16 && i_layout.feature() == 16) ||
+                (o_layout.format == format::b_fs_yx_fsv32 && i_layout.feature() == 32)) &&
+                i_layout.spatial(0) == o_layout.spatial(0) && i_layout.spatial(1) == o_layout.spatial(1) &&
+                i_layout.batch() == o_layout.batch()) {
+                r_node.can_be_optimized(true);
+                r_node.requires_reinterpret(true);
+            }
+
+            // Check redundant 1d reorder
+            size_t cnt = 0;
+            size_t size = 0;
+            for (auto dim : i_layout.get_dims()) {
+                if (dim == 1)
+                    cnt++;
+                else
+                    size = dim;
+            }
+
+            if ((i_layout.get_dims().size() == cnt + 1) && size == o_layout.count()) {
+                r_node.can_be_optimized(true);
+                r_node.requires_reinterpret(true);
+                continue;
+            }
+        }
+
```

Comment:
Could you please add some tests for optimization?

Reply:
Added